nav-challenge/src/components/containers/signUpContainer/SignUpContainer.tsx

```diff
@@ -0,0 +1,106 @@
+import moment from "moment";
+import React, { FC, useState } from "react";
+import AppProvider, {
+  appContext,
+  useDispatchApp,
+} from "../../../store/AppProvider";
+import store from "../../../store/Index";
+import { fetchRegisterUser } from "../../../store/user/UserActions";
+import { SignUpFormProps } from "../../../utils/types/SignUpFormProps";
+import { UserSignUpDataType } from "../../../utils/types/UserSignUpDataType";
+import SignUpPage from "../../pages/signUpPage/SignUpPage";
+
+const SignUpProvider: FC = () => {
+  return (
+    <AppProvider store={store} context={appContext}>
+      <SignUpContainer />
+    </AppProvider>
+  );
+};
+export const SignUpContext = React.createContext({} as SignUpFormProps);
+const SignUpContainer: FC = () => {
+  const userSignUpData: UserSignUpDataType = {
+    firstName: "",
+    lastName: "",
+    email: "",
+    username: "",
+    password: "",
+    dateOfBirth: "",
+  };
+
+  const userDispatch = useDispatchApp();
```

Comment:
Not sure what this is but there is a built-in react hook called `useDispatch` which can be used too